perceval/backends/mediawiki.py

```diff
@@ -0,0 +1,475 @@
+
+        logger.info("Total number of namespaces: %i", nns)
+        logger.info("Total number of pages: %i", npages)
+
+    def __build_page_reviews(self, page, reviews):
+        page['revisions'] = None
+        page['update'] = None
+        if str(page["pageid"]) in reviews["query"]["pages"]:
+            reviews_json = reviews["query"]["pages"][str(page["pageid"])]
+            if 'revisions' in reviews_json:
+                page["revisions"] = reviews_json['revisions']
+                page['update'] = self.__get_max_date(page['revisions'])
+        else:
+            logging.error("Revisions not found in %s", reviews["query"]["pages"])
+            logging.error("for page: %s", page)
+        return page
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the pages from the cache.
+
+        :returns: a generator of pages
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        pages_dicts = ['allrevisions','allpages','recentchanges']
+
+        # Namespaces -> pages -> revisions
+        next(cache_items) # First namespaces, we don't need them
+        for items in cache_items:
+            data_json = json.loads(items)
+            for pages_dict in pages_dicts:
+                if pages_dict in data_json['query']:
+                    pages_json = data_json['query'][pages_dict]
+                    for page in pages_json:
+                        page_reviews = self.__build_page_reviews(page, json.loads(next(cache_items)))
+                        yield page_reviews
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Build item."""
+        return str(item['pageid'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MediaWiki item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        updated = None
+        if item['update']:
+            updated = float(item['update'])
+        return updated
+
+class MediaWikiClient:
+    """MediaWiki API client.
+
+    This class implements a simple client to retrieve pages from
+    projects in a MediaWiki node.
+
+    :param url: URL of jenkins node: https://build.opnfv.org/ci
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, url):
+        self.url = url
+        self.api_url = self.url + "/" + "api.php"
+
+    def get_namespaces(self):
+        """ Retrieve all contents namespaces
+        """
+        ns_query = "action=query&meta=siteinfo&siprop=namespaces"
```

Comment:
The parameters attributes should be build using `requests` package. See how we do it on [Bugzilla client](https://github.com/grimoirelab/perceval/blob/master/perceval/backends/bugzilla.py#L550) and on its [call method](https://github.com/grimoirelab/perceval/blob/master/perceval/backends/bugzilla.py#L588).

You should change this in the rest of the code.